Implement the warning for typeclass instances without an explicit locality. (#14208)
coq/coq: doc/changelog/07-vernac-commands-and-options/14208-warn-unqualified-typeclass-instance.rst

- **Deprecated:**

>   Similarly to hints, typeclass instances added outside of sections
>   without an explicit locality now generate a deprecation warning. See
>   the hint documentation for more background

Reviewer: ```suggestion
  Like hints, typeclass instances added outside of sections
  without an explicit locality now generate a deprecation warning. See
  :ref:`XXXXXXX`
```

The hyperlink should go to type-classes.rst, presumably to the text you add there.  It shouldn't mention "hint documentation"

Author: Done.
